src/chocolatey/infrastructure/commands/IListCommand.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright © 2015 - Present RealDimensions Software, LLC
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// 
+// You may obtain a copy of the License at
+// 
+// 	http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace chocolatey.infrastructure.commands
+{
+    using System.Collections.Generic;
+    using app.configuration;
+
+    public interface IListCommand<out T> : ICommand
+    {
+        IEnumerable<T> list(ChocolateyConfiguration config, bool logResults = true);
```

Comment:
> Either way it ends up requiring us to rewrite other things to respect that new setting, or else the setting doesn't do anything most of the time -- both options seemed undesirable to me.

There is only one or two places where this would be used right now, all related to lists. As far as what can call the setting, most folks are going to be exposed to what the command interfaces tell them they can set, not the configuration.